addons/l10n_in/models/account_invoice.py

```diff
@@ -105,7 +105,7 @@ def _post(self, soft=True):
                 move.l10n_in_state_id = company_unit_partner.state_id
 
             shipping_partner = move._l10n_in_get_shipping_partner()
-            move.l10n_in_gstin = move._l10n_in_get_shipping_partner_gstin(shipping_partner)
+            move.l10n_in_gstin = move._l10n_in_get_shipping_partner_gstin(shipping_partner) or move.partner_id.vat
```

Comment:
@jva-odoo And why don't we put it in the _l10n_in_get_shipping_partner_gstin logic?

Reply:
because this is a model type method and we overwrite in l10n_in_sale

Reply:
You have to configure things awfully bad in that case, because this means that the shipping partner has no vat and no gstin, then fall back on the invoicing partner's vat and not the gstin, which should already have the same vat if they are configured as children...  I really wonder what the functional use case is and you are not describing it.  @jva-odoo

Reply:
@jco-odoo shipping partner is not a child of invoice partner because of Bill to Ship to Trangesction. 

And We also have a plan to remove this in the future.

Reply:
But I would suspect that if he adds an extra partner independent of that, he would have thought about adding the proper vat/gstin...  @jva-odoo

Reply:
And the system gives an error, so he is aware of it...

Reply:
I think the same, but here case where the shipping partner is unregistered and the customer can't create this type of invoice.

Reply:
And it can not be the gstin field on the move.partner_id instead?  And is not it more logical to put this logic in the _l10n_in_get_shipping_partner immediately or is it used elsewhere for the delivery address?

Reply:
_l10n_in_get_shipping_partner is api.model method so I can't use self.partner_id so I am put logic in _l10n_in_get_shipping_partner then I need to add other parameter but this break stable version policy.
It not used in any other model and module.
It's just overwritten in l10n_in_sale

Reply:
@jva-odoo It is only the _gstin method that is api.model

Reply:
@jco-odoo ok get your point, If I do this logic in _l10n_in_get_shipping_partner then the result is wrong as per method name because if shipping_partner has GSTN then it returns the same shipping partner else invoice partner so it's wrong as per method function.